Sources/SwiftProtobuf/BinaryEncoder.swift

```diff
@@ -82,63 +80,41 @@ public struct BinaryEncoder {
         append(value ? 1 : 0)
     }
 
-    mutating func putFixedUInt64(value : UInt64) {
-        var v = value
+    mutating func putFixedUInt64(value: UInt64) {
+        var v = value.littleEndian
         let n = MemoryLayout<UInt64>.size
-        withUnsafePointer(to: &v) { v -> () in
-            v.withMemoryRebound(to: UInt8.self, capacity: n) { p -> () in
-                let buff = UnsafeBufferPointer<UInt8>(start: p, count: n)
-                append(contentsOf: buff)
-            }
-        }
+        memcpy(pointer, &v, n)
+        pointer = pointer.advanced(by: n)
     }
 
-    mutating func putFixedUInt32(value : UInt32) {
-        var v = value
+    mutating func putFixedUInt32(value: UInt32) {
+        var v = value.littleEndian
         let n = MemoryLayout<UInt32>.size
-        withUnsafePointer(to: &v) { v -> () in
-            v.withMemoryRebound(to: UInt8.self, capacity: n) { p -> () in
-                let buff = UnsafeBufferPointer<UInt8>(start: p, count: n)
-                append(contentsOf: buff)
-            }
-        }
+        memcpy(pointer, &v, n)
+        pointer = pointer.advanced(by: n)
     }
 
     mutating func putFloatValue(value: Float) {
         var v = value
```

Comment:
I had thought that all modern processors used the same endianness for floating point, but I dug a little more -- it appears I was wrong about that.

So I expanded this a bit:  It now reads into a correspondingly-sized `UInt`, then swaps to native endianness, then copies to a `Float` or `Double`.

Fixed.